marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mastering.sjs

```diff
@@ -0,0 +1,102 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+const DataHubSingleton = require('/data-hub/5/datahub-singleton.sjs');
+
+// define constants for caching expensive operations
+const dataHub = DataHubSingleton.instance();
+
+const collections = ['http://marklogic.com/data-hub/steps/mastering', 'http://marklogic.com/data-hub/steps'];
+const databases = [dataHub.config.STAGINGDATABASE, dataHub.config.FINALDATABASE];
+const permissions = [xdmp.permission(dataHub.consts.DATA_HUB_FLOW_WRITE_ROLE, 'update'), xdmp.permission(dataHub.consts.DATA_HUB_FLOW_READ_ROLE, 'read')];
+const requiredProperties = ['name'];
+
+function getNameProperty() {
+    return 'name';
+}
+
+function getEntityNameProperty() {
+    return 'targetEntityType';
+}
+
+function getSelectedSourceProperty() {
+    return 'selectedSource';
+}
+
+function getVersionProperty() {
+    return null;
+}
+
+function getCollections() {
+    return collections;
+}
+
+function getStorageDatabases() {
+    return databases;
+}
+
+function getPermissions() {
+    return permissions;
+}
+
+function getFileExtension() {
+  return '.step.json';
+}
+
+function getDirectory() {
+  return "/steps/mastering/";
+}
+
+function getArtifactNode(artifactName, artifactVersion) {
+    const results = cts.search(cts.andQuery([cts.collectionQuery(collections[0]), cts.jsonPropertyValueQuery('name', artifactName)]));
+    return fn.head(results);
+}
+
+function validateArtifact(artifact) {
+    const missingProperties = requiredProperties.filter((propName) => !artifact[propName]);
```

Comment:
I don't think any validation is required here - it's best just to return the artifact.

Reply:
Removed validation